src/aws_lambda_python/mpic_coordinator/cohort_creator.py

```diff
@@ -0,0 +1,132 @@
+from importlib import resources
+from itertools import cycle, chain
+import random
+import yaml
+from pydantic import TypeAdapter
+
+from aws_lambda_python.mpic_coordinator.domain.remote_perspective import RemotePerspective
+
+
+class CohortCreator:
+    @staticmethod
+    def build_randomly_shuffled_available_perspectives_per_rir(available_perspectives: list[str], random_seed: bytes) -> dict[str, list[RemotePerspective]]:
+        # convert available_perspectives to a list of RemotePerspective objects
+        remote_perspectives = []
+        all_possible_perspectives_by_code = CohortCreator.load_aws_region_config()
+
+        for perspective in available_perspectives:
+            perspective_rir = perspective.split('.')[0]
+            perspective_code = perspective.split('.')[1]
+
+            if perspective_code not in all_possible_perspectives_by_code.keys():
+                continue  # TODO throw an error? check this case in the validator?
+            else:
+                full_perspective = all_possible_perspectives_by_code[perspective_code]
+                # TODO discuss: do we even need RIRs specified in the input? code should be unique enough
+                remote_perspectives.append(full_perspective)
+
+        random.seed(random_seed)
+        random.shuffle(remote_perspectives)
+        perspectives_per_rir = {}
+        for perspective in remote_perspectives:
+            if perspective.rir not in perspectives_per_rir:
+                perspectives_per_rir[perspective.rir] = []
+            perspectives_per_rir[perspective.rir].append(perspective)
+
+        return perspectives_per_rir
+
+    @staticmethod
+    def load_aws_region_config():
+        """
+        Reads in the available AWS regions from a configuration yaml and returns them as a list.
+        :return: list of available AWS regions
+        """
+        with resources.open_text('resources', 'aws_region_config.yaml') as file:
+            aws_region_config_yaml = yaml.safe_load(file)
+            aws_region_type_adapter = TypeAdapter(list[RemotePerspective])
+            aws_regions_list = aws_region_type_adapter.validate_python(aws_region_config_yaml['aws_available_regions'])
+            aws_regions_dict = {region.code: region for region in aws_regions_list}
+            return aws_regions_dict
+
+    @staticmethod
+    def create_perspective_cohorts(perspectives_per_rir: dict, cohort_size: int):
+        if cohort_size == 1:
+            return [[region] for region in
+                    chain.from_iterable(perspectives_per_rir.values())]  # TODO limit cohort number in this case?
+        elif len(perspectives_per_rir.keys()) < 2:  # else if only one rir, can't meet requirements
+            return []  # TODO throw an error? check this case in the validator?
```

Comment:
Makes sense. My thinking is, since distinct RIR regions are a BR requirement, perhaps it's best to have such a flag be associated with "diagnostic" mode, similar to other configuration that would normally invalidate the corroboration.
That said, for what it's worth, if diagnostics mode removes the RIR requirement (or distance requirement for that matter), that changes the algorithm here and there, as well. The diagnostic mode "feature" is becoming a non-trivial piece of work so I'm probably going to postpone work on it until I can get this codebase more complete in terms of required behavior (i.e., ACME, better HTTP challenge, etc.).